dotnet/src/SemanticKernel.Skills/Skills.Grpc/GrpcOperationRunner.cs

```diff
@@ -0,0 +1,253 @@
+    /// <param name="operation">The gRPC operation.</param>
+    /// <param name="arguments">The gRPC operation arguments.</param>
+    /// <returns>The channel address.</returns>
+    private string GetAddress(GrpcOperation operation, IDictionary<string, string> arguments)
+    {
+        var address = default(string);
+
+        if (!arguments.TryGetValue(AddressArgumentName, out address))
+        {
+            address = operation.Address;
+        }
+
+        if (string.IsNullOrEmpty(address))
+        {
+            throw new GrpcOperationException($"No address provided for the '{operation.Name}' gRPC operation.");
+        }
+
+        return address!;
+    }
+
+    /// <summary>
+    /// Creates a marshaller - a typed abstraction for gRPC message serialization and deserialization.
+    /// </summary>
+    /// <param name="contractType">The message contract data type.</param>
+    /// <returns>The marshaller.</returns>
+    private Marshaller<T> CreateMarshaller<T>(Type contractType)
+    {
+        byte[] Serialize(T instance)
+        {
+            using var memoryStream = new MemoryStream();
+
+            Serializer.NonGeneric.Serialize(memoryStream, instance);
+
+            return memoryStream.ToArray();
+        }
+
+        T Deserialize(byte[] source)
+        {
+            using var memoryStream = new MemoryStream(source);
+
+            return (T)Serializer.NonGeneric.Deserialize(contractType, memoryStream);
+        }
+
+        return Marshallers.Create((instance) => Serialize(instance), (bytes) => Deserialize(bytes));
+    }
+
+    /// <summary>
+    /// Creates a gRPC operation request.
+    /// </summary>
+    /// <param name="operation">The gRPC operation.</param>
+    /// <param name="type">The operation request data type.</param>
+    /// <param name="arguments">The operation arguments.</param>
+    /// <returns>The operation request instance.</returns>
+    private object GenerateOperationRequest(GrpcOperation operation, Type type, IDictionary<string, string> arguments)
+    {
+        var instance = Activator.CreateInstance(type);
+
+        foreach (var field in operation.Request.Fields)
+        {
+            if (!arguments.TryGetValue(field.Name, out var value))
+            {
+                throw new GrpcOperationException($"No argument found for field - '{field.Name}' for operation - '{operation.Name}'");
+            }
+
+            var pascalCaseName = CultureInfo.InvariantCulture.TextInfo.ToTitleCase(field.Name);
+
+            type.GetProperty(pascalCaseName).SetValue(instance, value);
+        }
+
+        return instance;
+    }
+
+    /// <summary>
+    /// Builds gRPC operation data contract type.
+    /// </summary>
+    /// <param name="dataContractMetadata">The data contract type metadata.</param>
+    /// <returns>.NET type representing the data contract type.</returns>
+    public static Type BuildGrpcOperationDataContractType(GrpcOperationDataContractType dataContractMetadata)
```

Comment:
Very cool! I've been wanting to use Ref.Emit in SK for a while. Glad to see... more applications to come, I'm sure.